keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signers/hashicorp/dsl/HashicorpNode.java

```diff
@@ -12,86 +12,293 @@
+    if (vaultInstallDir == null) {
+      throw new IllegalStateException("System property [vaultInstallDir] is missing.");
+    }
+
+    final Path vault = Path.of(vaultInstallDir).resolve(vaultBinary);
+    if (!vault.toFile().exists()) {
+      throw new IllegalStateException("Vault binary doesn't exist. " + vault.toString());
+    }
 
-  public static HashicorpNode createAndStartHashicorp(
-      final DockerClient dockerClient, final boolean withTls) {
     try {
       final HashicorpNode hashicorpNode =
           withTls
-              ? new HashicorpNode(dockerClient, SelfSignedCertificate.generate())
-              : new HashicorpNode(dockerClient);
+              ? new HashicorpNode(vault, SelfSignedCertificate.generate())
+              : new HashicorpNode(vault, null);
       hashicorpNode.start();
       return hashicorpNode;
     } catch (final Exception e) {
       throw new RuntimeException("Failed to create Hashicorp Node.", e);
     }
   }
 
-  private void start() {
-    final HashicorpVaultDockerCertificate hashicorpVaultDockerCertificate =
-        serverTlsCertificate.map(HashicorpVaultDockerCertificate::create).orElse(null);
+  private HashicorpNode(final Path vaultPath, final SelfSignedCertificate serverTlsCertificate) {
+    this.vaultPath = vaultPath;
+    this.serverTlsCertificate = Optional.ofNullable(serverTlsCertificate);
+  }
+
+  void start()
+      throws IOException, CertificateEncodingException, TimeoutException, InterruptedException {
+    startVaultServer();
+    hashicorpVaultCommands = new HashicorpVaultCommands(vaultPath.toString(), vaultUrl());
 
-    hashicorpVaultDocker =
-        HashicorpVaultDocker.createVaultDocker(dockerClient, hashicorpVaultDockerCertificate);
-    Runtime.getRuntime().addShutdownHook(new Thread(this::shutdown));
+    final HashicorpVaultTokens hashicorpVaultTokens = initVault();
+    unseal(hashicorpVaultTokens.getUnsealKey());
+    enableHashicorpKeyValueV2Engine();
   }
 
-  public synchronized void shutdown() {
-    if (hashicorpVaultDocker != null) {
-      hashicorpVaultDocker.shutdown();
-      hashicorpVaultDocker = null;
+  // Start vault server. Sets port variable once started.
+  private void startVaultServer() throws IOException, CertificateEncodingException {
+    LOG.info("Starting vault server");
+    final AtomicBoolean serverStarted = new AtomicBoolean(false);
+    serverFuture =
+        new ProcessExecutor(vaultPath.toString(), "server", "-config=" + getVaultServerConfig())
+            .readOutput(true)
+            .redirectOutput(
+                new LogOutputStream() {
+                  @Override
+                  protected void processLine(final String line) {
+                    LOG.info(line);
+
+                    // process Vault port (i.e. cluster port - 1)
+                    if (line.contains("cluster address")) {
+                      final Matcher matcher = portPattern.matcher(line);
+                      matcher.find();
```

Comment:
Is find needed? the result of the find isn't being used

Reply:
yes, usually this comes in an `if` condition, however, we have already used `contains` above so it is guaranteed to return true.

Reply:
do we need to do the contains? Or just use the matcher?

Reply:
let me fix it :)

Reply:
fixed